integrations/gen_doc_collector_page.py

```diff
@@ -0,0 +1,75 @@
+import json
+
+"""
+This script reads the integrations/integrations.js file and generates the list of data collection integrations inside collectors/COLLECTORS.md, with proper links that Learn can replace into Learn links.
+"""
+
+# Open integrations/integrations.js and extract the dictionaries
+with open('integrations/integrations.js') as dataFile:
+    data = dataFile.read()
+
+    categories_str = data.split("export const categories = ")[1].split("export const integrations = ")[0]
+    integrations_str = data.split("export const categories = ")[1].split("export const integrations = ")[1]
+
+    categories = json.loads(categories_str)
+    integrations = json.loads(integrations_str)
+
+cat_dict = {}
+data_col_cat = {}
+
+
+def recursive(categories):
+    for category in categories:
+        data_col_cat[category['id']] = category['name']
+        if category['children']:
+            recursive(category['children'])
+
+
+recursive(categories[1]['children'])
+
+
+def construct_dict(array, integration):
+    for element in array:
+        try:
+            cat_dict[element].append(integration)
+        except KeyError:
+            cat_dict[element] = []
+            cat_dict[element].append(integration)
+
+
+md = ""
+
+for integration in integrations:
+    try:
+        construct_dict(integration['meta']['monitored_instance']['categories'], integration)
+    except KeyError:
+        pass
+
+sorted_dict = dict(sorted(cat_dict.items()))
+
+
+for category_id in sorted_dict:
+    heading = '#' * len(category_id.split('.'))
+
+    for cat in data_col_cat:
+        if cat == category_id:
+            name = data_col_cat[cat]
+
+    md += f'#{heading} {name}\n\n\n'
+    names = []
+    for integration in sorted_dict[category_id]:
+        name = integration['meta']['monitored_instance']['name']
+        link = integration['edit_link'].replace("metadata.yaml", "") + \
+            "integrations/" + name.lower().replace(" ", "_").replace("/", "-")+".md"
+        names.append(f"[{name}]({link})")
+    for integration_name in sorted(names):
+        md += "- " + integration_name + "\n\n"
+
+
+with open("collectors/COLLECTORS.md", "r+") as txt:
+    wholefile = txt.read()
+    output = wholefile.split("## Available Data Collection Integrations")[0]
+
+    output += "## Available Data Collection Integrations\n<!-- AUTOGENERATED PART BY integrations/gen_doc_collector_page.py SCRIPT, DO NOT EDIT MANUALLY -->\n" + md
+    txt.seek(0)
+    txt.write(output)
```

Comment:
Possibly consider rewriting this to use pathlib. For example:

```suggestion
outfile = pathlib.Path("./collectors/Collectors.md")
output = outfile.read_text().split("## Available Data Collection Integrations")[0]
output += "## Available Data Collection Integrations\n<!-- AUTOGENERATED PART BY integrations/gen_doc_collector_page.py SCRIPT, DO NOT EDIT MANUALLY -->\n" + md
outfile.write_text(output)
```

Doing so will require importing `pathlib` near the top, but the result is slightly easier to read, and should be a tiny bit faster and more memory efficient.